fv3core/stencils/d2a2c_vect.py

```diff
@@ -382,72 +379,110 @@ class DGrid2AGrid2CGridVectors:
 
-        is2 = npt + OFFSET if self.grid.west_edge else self.grid.is_ - 1
-        ie2 = nx - npt if self.grid.east_edge else self.grid.ie + 1
-        js2 = npt + OFFSET if self.grid.south_edge else self.grid.jsd
-        je2 = ny - npt if self.grid.north_edge else self.grid.jed
+        is2 = npt + OFFSET if grid_indexing.west_edge else grid_indexing.isc - 1
+        ie2 = nx - npt if grid_indexing.east_edge else grid_indexing.iec + 1
+        js2 = npt + OFFSET if grid_indexing.south_edge else grid_indexing.jsd
+        je2 = ny - npt if grid_indexing.north_edge else grid_indexing.jed
 
-        ifirst = self.grid.is_ + 2 if self.grid.west_edge else self.grid.is_ - 1
-        ilast = self.grid.ie - 1 if self.grid.east_edge else self.grid.ie + 2
+        ifirst = (
+            grid_indexing.isc + 2 if grid_indexing.west_edge else grid_indexing.isc - 1
+        )
+        ilast = (
+            grid_indexing.iec - 1 if grid_indexing.east_edge else grid_indexing.iec + 2
+        )
         idiff = ilast - ifirst + 1
 
-        jfirst = self.grid.js + 2 if self.grid.south_edge else self.grid.js - 1
-        jlast = self.grid.je - 1 if self.grid.north_edge else self.grid.je + 2
+        jfirst = (
+            grid_indexing.jsc + 2 if grid_indexing.south_edge else grid_indexing.jsc - 1
+        )
+        jlast = (
+            grid_indexing.jec - 1 if grid_indexing.north_edge else grid_indexing.jec + 2
+        )
         jdiff = jlast - jfirst + 1
 
-        js3 = npt + OFFSET if self.grid.south_edge else self.grid.jsd
-        je3 = ny - npt if self.grid.north_edge else self.grid.jed
-        jdiff3 = je3 - js3 + 1
-
         self._set_tmps = FrozenStencil(
             func=set_tmps,
-            origin=self.grid.full_origin(),
-            domain=self.grid.domain_shape_full(),
+            origin=grid_indexing.origin_full(),
+            domain=grid_indexing.domain_full(),
         )
 
         self._lagrange_interpolation_y_p1 = FrozenStencil(
             func=lagrange_interpolation_y_p1,
             origin=(is1, js1, 0),
-            domain=(ie1 - is1 + 1, je1 - js1 + 1, self.grid.npz),
+            domain=(ie1 - is1 + 1, je1 - js1 + 1, grid_indexing.domain[2]),
         )
 
         self._lagrange_interpolation_x_p1 = FrozenStencil(
             func=lagrange_interpolation_x_p1,
             origin=(is2, js2, 0),
-            domain=(ie2 - is2 + 1, je2 - js2 + 1, self.grid.npz),
+            domain=(ie2 - is2 + 1, je2 - js2 + 1, grid_indexing.domain[2]),
         )
 
-        origin = self.grid.full_origin()
-        domain = self.grid.domain_shape_full()
-        ax_offsets = axis_offsets(self.grid, origin, domain)
-        if namelist.npx <= 13 and namelist.layout[0] > 1:
+        origin = grid_indexing.origin_full()
+        domain = grid_indexing.domain_full()
+        ax_offsets = axis_offsets(grid_indexing, origin, domain)
+        if npx <= 13 and layout[0] > 1:
```

Comment:
This was written to represent the use case we wanted to avoid. But actually for now, you could just do exactly what the fortran is doing with GT4PY_DEV (and include a TODO to revisit still). 
npt = 0 if npt > (je - js) .or. npt > (ie - is)
 That is already set in this code above. So the check here could just be
`if npt == 0` set d2a2c_avg_offset = -1 otherwise   d2a2c_avg_offset = 3
This may not be the right thing for nested, but a lot of other code also would need to be added/modified to support nested. Then you can ignore the npx and layout check. I think this ended up this way due to multiple refactors and rollbacks by multiple people.